src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -0,0 +1,92 @@
+module.exports = {
+  beforeCreate(event) {
+    console.log("**** beforeCreate hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeCreateMany(event) {
+    console.log("**** beforeCreateMany hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeUpdate(event) {
+    console.log("**** beforeUpdate hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+
+  beforeUpdateMany(event) {
+    console.log("**** beforeUpdateMany hook is activated! ****");
+    const eventData = event.params.data;
+    runMetaTest(eventData);
+  },
+};
+
+async function runMetaTest(eventData) {
+  const tests = await getInternalTests(eventData);
+  const metaTests = await getMetaTests(eventData);
+
+  tests.forEach((test) =>
+    console.log("final test.internalTest: ", test.internalTest)
+  );
+
+  metaTests.forEach((metaTest) =>
+    console.log("final metaTest caseCode: ", metaTest.caseCode)
+  );
+
+  return false;
+}
+
+async function getInternalTests(eventData) {
+  // Original Finding Internal Test
+  const internalTests = await strapi.db
+    .query("challenge.code-challenge-test")
+    .findMany({
+      select: ["id", "internalTest"],
+      where: {
+        id: {
+          $gte: getCurrentTestID(eventData),
+        },
+      },
+    });
+
+  console.log("internalTests: ", internalTests);
+  return internalTests;
+}
+
+function getCurrentTestID(eventData) {
```

Comment:
I don't really understand this and `getMetaTestId`. Why are you getting the minimum value? I can't intuitively guess why

Reply:
Also, it's a bad practice to have it receive the entire eventData. first of all, eventData is structurally different between `beforeCreate` and `beforeUpdate`, so you're running the risk of someone working on one of the lifecycles breaking the other. Second, it makes it harder to test this in unit tests because then you need to mock the entire massive object. In this case, just pass in the tests

Reply:
> I don't really understand this and `getMetaTestId`. Why are you getting the minimum value? I can't intuitively guess why

I will add a comment to make it more understandable. I ran into an problem where multiple tests/metatests were being returned out of order. 

I needed a function Math.min(...test.IDs) to sort them by lowest ID --> highest

problem IDs (30, 29, 31)
after function (29, 30, 31)

Reply:
> Also, it's a bad practice to have it receive the entire eventData. first of all, eventData is structurally different between `beforeCreate` and `beforeUpdate`, so you're running the risk of someone working on one of the lifecycles breaking the other. Second, it makes it harder to test this in unit tests because then you need to mock the entire massive object. In this case, just pass in the tests

Okay I see what you mean here. 

I didn't realize the event object would be changing. How do you know this? 

How long have you been working with strapi? 

Learning Strapi feels almost a little like learning react but for node...maybe this is why they call it a framework...

Reply:
It's all patterns. You'll recognize many similarities with other frameworks. Strapi isn't very unique or anything

Reply:
"I didn't realize the event object would be changing. How do you know this?" event objects are usually not standardized I think. Like even DOM events i would expect to change. In typescript you have a whoel bunch of different types for DOM events